spicy/toolchain/src/compiler/codegen/grammar-builder.cc

```diff
@@ -19,14 +19,14 @@ using hilti::util::fmt;
 namespace {
 
 struct Visitor : public hilti::visitor::PreOrder<Production, Visitor> {
-    Visitor(codegen::GrammarBuilder* gb, Grammar* g) : gb(gb), grammar(g) {}
+    Visitor(CodeGen* cg, codegen::GrammarBuilder* gb, Grammar* g) : cg(cg), gb(gb), grammar(g) {}
+    CodeGen* cg;
     codegen::GrammarBuilder* gb;
     Grammar* grammar;
 
     using CurrentField = std::pair<const spicy::type::unit::item::Field&, std::reference_wrapper<hilti::Node>&>;
     std::vector<CurrentField> fields;
     hilti::util::Cache<std::string, Production> cache;
-    hilti::util::Uniquer<std::string> uniquer;
```

Comment:
We have another (`static`!) `Uniquer` in `type::unit::item::Field`. Do you think we could clean that one up as well as part of this work so we have uniquing always as a part of cg?

Reply:
Don't see an easy an easy way to unify those as field doesn't have access to a codegen. That's ok: if it only needs to be unique across fields, it's ok to leave there.